cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -143,40 +134,91 @@ func (c *Controller) getSpcResource(key string) (*apis.StoragePoolClaim, error)
 	return spcGot, nil
 }
 
+// synSpc is the function which tries to converge to a desired state for the spc.
 func (c *Controller) syncSpc(spcGot *apis.StoragePoolClaim) error {
-	if len(spcGot.Spec.Disks.DiskList) > 0 {
-		// TODO : reconciliation for manual storagepool provisioning
-		glog.V(1).Infof("No reconciliation needed for manual provisioned pool of storagepoolclaim %s", spcGot.Name)
-		return nil
-	}
 	glog.V(1).Infof("Syncing storagepoolclaim %s", spcGot.Name)
-	// Get kubernetes clientset
-	// namespaces is not required, hence passed empty.
-	newK8sClient, err := k8s.NewK8sClient("")
+	currentPoolCount, err := c.getCurrentPoolCount(spcGot)
 	if err != nil {
 		return err
 	}
-	// Get openebs clientset using a getter method (i.e. GetOECS() ) as
-	// the openebs clientset is not exported.
-	newOecsClient := newK8sClient.GetOECS()
-
-	// Get the current count of provisioned pool for the storagepool claim
-	spList, err := newOecsClient.OpenebsV1alpha1().StoragePools().List(metav1.ListOptions{LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spcGot.Name})
-	if err != nil {
-		return fmt.Errorf("unable to list storagepools: %v", err)
-	}
-	currentPoolCount := len(spList.Items)
-
-	// If current pool count is less than maxpool count, try to converge to maxpool
-	if currentPoolCount < int(spcGot.Spec.MaxPools) {
+	maxPoolCount := int(spcGot.Spec.MaxPools)
```

Comment:
Since manual pool SPC didn't have this in 0.8, test if the upgrade will cause a failure here?

Reply:
It should not cause but I will test it.